### PR TITLE
Argo bridge plugin support

### PR DESCRIPTION
**Pull request by jacobsee**
### What does this PR do?
Adds support for plugins in the GitLab/Argo bridge, enabling preprocessing of GitLab projects before adding them to Argo (or choosing not to).

### Is there a relevant Issue open for this?
no

### People to notify
cc: @oybed @pabrahamsson

### CODE REVIEW

File: argo-cd-bridges/gitlab/deploy/cronjob.yml [hunk 1]
```diff
@@ -22,6 +22,13 @@ spec:
             resources: {}
             terminationMessagePath: /dev/termination-log
             terminationMessagePolicy: File
+            volumeMounts:
+              - name: plugin-volume
+                mountPath: /plugins
+          volumes:
+            - name: plugin-volume
+              configMap:
+                name: gitlab-argo-bridge-plugins
```
Comment (pabrahamsson) [inline argo-cd-bridges/gitlab/deploy/cronjob.yml:31]: Can you add an example to the README showing the format of this configMap?

**Review by pabrahamsson — CHANGES_REQUESTED** [review]
Overall looks great, one minor comment inline.

**Review by oybed — COMMENTED** [review]
This looks great to me. +1 to @pabrahamsson 's comment.

Side-comment: Maybe this is a follow-up change, but could we make it so the "plugin framework" supports Ansible execution? ... or would that be a separate "plugin" to handle? (i.e.: a python plugin that's an "ansible executor")

File: argo-cd-bridges/gitlab/deploy/cronjob.yml (hunk 1 above)
Comment (jacobsee) [reply argo-cd-bridges/gitlab/deploy/cronjob.yml:31]: Yep - just added!

**Comment by jacobsee** [comment]
@oybed I was _imagining_ anything more complicated than this to be a plugin that knows how to handle more situations, yes, but... I also don't know how we plan to use Ansible here, so might deserve a conversation about what's needed.

**Review by oybed — APPROVED** [review]
LGTM

**Pull request merged**